src/IntelliTect.Coalesce/TypeDefinition/TypeViewModel.cs

```diff
@@ -46,7 +46,7 @@ internal TypeViewModel(ReflectionRepository? reflectionRepository) : this()
         /// Excludes <see cref="string"/> and <see cref="T:byte[]"/>
         /// </para>
         /// </summary>
-        public bool IsCollection => IsA<IEnumerable>() && !IsString && !IsByteArray;
+        public bool IsCollection => IsA(typeof(IEnumerable<>)) && !IsString && !IsByteArray;
```

Comment:
Do we still care to check for older (non-generic) collection types, or is the assumption that everything we care about will have generic support?

Reply:
No, we don't care. If it isn't generic, the generated code that does the mapping in the DTOs wouldn't work. Arguably this shouldn't be checking IEnumerable at all and should indeed be checking ICollection, but legacy prevails.